src/OpenTelemetry.Contrib.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs

```diff
@@ -42,9 +46,33 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
-            activity.DisplayName = this.GetDisplayName(activity);
+            string displayName = this.GetDisplayName(activity);
+            ActivityContext activityContext = default;
+            Activity originalActivityCurrent = Activity.Current;
+            if (activity.ParentSpanId == default)
+            {
+                Activity.Current = null;
+            }
+            else
+            {
+                activityContext = new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true);
```

Comment:
Could we use `activityContext = activity.Context` here? Or is it mandatory to create a new context here?

Reply:
@cijothomas , do u know what would be the difference between using activity.Context and creating our own? I remember that you suggested this in the AspNetCore migration as well.

Reply:
@alexvaluyskiy , i checked the values:
![image](https://user-images.githubusercontent.com/16199012/103213785-82923100-48ed-11eb-9f8e-604f7d8e2e51.png)

Reply:
The new activity being created must be a "sibling" of the original activity. This is achieved by setting the parent of the original activity as the parent of the to-be-created activity. Using activity.Context will make the new activity as a child of the original activity, instead of sibling.